modules/cnv_install_1_4.adoc

```diff
@@ -0,0 +1,53 @@
+[[install_cnv_apb]]
+=== Installing {ProductName} to {product-title}
+
+The `kubevirt-ansible` RPM contains the latest automation for deploying 
+{ProductName}. This procedure installs all {ProductName} components to your 
+{product-title} cluster. 
+
+This procedure installs the following components:
+
+* {ProductName} core components (KubeVirt)
+* Containerized data importer (CDI) controller
+* Multus and Open vSwitch (OVS) container network interface plug-ins
+* Updated {product-title} web console
+
+.Prerequisites
+* {product-title} 3.11 cluster, running
+* User with `cluster-admin` privileges
+* xref:../apb_devel/cli_tooling.adoc#apb-devel-cli[APB tool]
+* `virtctl` client utility, installed
+
+.Procedure
+
+. Install the `kubevirt-ansible` RPM and its dependencies:
++
+----
+$ yum install kubevirt-ansible
+----
+
+. Navigate to the `kubevirt-ansible` directory:
++
+----
+$ cd /usr/share/ansible/kubevirt-ansible
+----
+
+. Edit `cnv.yml` to configure `kubevirt-ansible`:
++
+----
+$ vim vars/cnv.yml 
```

Comment:
@ousleyp please do not navigate to `/usr/share/ansible/kubevirt-ansible` and edit the original `vars/cnv.yml` file.
We should probably ask to copy it in the local path (E.g. `cp /usr/share/ansible/kubevirt-ansible/vars/cnv.yml cnv-vars.yml` or something similar). Then it means you have to change also the `vim` and `ansible-playbook` commands accordingly, something like: `ansible-playbook -e @cnv-vars.yml /usr/share/ansible/kubevirt-ansible/playbooks/kubevirt.yml`

Reply:
We clearly should not edit an rpm-shipped file, but why not `cd` to shorten our paths? A user should be encourage to add her changes to a homedir file, and add `-e @~/local-cnv-vars.yml` to `-e @vars/all.yml -e @vars/cnv.yml`

Also, iirc, we should include vars/all.yml in the commandline.

I would make sure that mentioning `~/local-cnv-args.yml` is not on the main procedure. We should be working fine out of the box. Only users with special needs (e.g. off-internet) need to have this file.

Reply:
New proposed steps, based on feedback from @simon3z and @dankenigsberg (let me know how it looks):

- `yum install kubevirt-ansible`
- `cp /usr/share/ansible/kubevirt-ansible/vars/cnv.yml local-cnv-vars.yml`
- `vim local-cnv-vars.yml`
- `cd /usr/share/ansible/kubevirt-ansible`
- `ansible-playbook -e @~/local-cnv-vars.yml -e @vars/all.yml -e @vars/cnv.yml playbooks/kubevirt.yml`
- verify install by navigating to the web UI

Feedback also welcome from @rthallisey (and anyone else).

Reply:
I think that the last `-e` wins, so `@~/local-cnv-vars.yml` should come last. If it is really a *copy* of cnv.yml,  you do not need cnv.yml.

you could say: "in case you need to change anything in cnv.yml, copy it to ~/cnv.yml, modify it and use -e ~/cnv.yml instead of -e vars/cnv.yml"

the important bit is to keep the plain installation short and sweet. I hope that most users would not need to edit cnv.yml.

Reply:
> but why not `cd` to shorten our paths?

@dankenigsberg I go by the principle to not leave the session *dirty* at the end of the documented steps.

There may be other documented actions in the next chapters that do not assume you're in `/usr/share/ansible/kubevirt-ansible`.

Reply:
How about proper bash programming with

`pushd <dir>`
do your things
`popd`